dist/tools/pyterm/pyterm

```diff
@@ -245,16 +248,27 @@ class SerCmd(cmd.Cmd):
                 sys.exit(1)
         # otherwise go for the serial port
         elif self.port:
+            connected = False
             self.logger.info("Connect to serial port %s" % self.port)
-            try:
-                self.serial_connect()
-            except OSError as e:
-                self.logger.error("Cannot connect to serial port {}: {}"
-                                  .format(self.port, e.strerror))
-                sys.exit(1)
-            except serial.SerialException as e:
-                self.logger.error("%s", e.strerror)
-                sys.exit(1)
+            while not connected:
+                try:
+                    self.serial_connect()
+                except OSError as e:
```

Comment:
This doesn't work on my machine: OSError is catched before serial.SerialException. The following works:
```suggestion
                except serial.SerialException as e:
                    self.logger.error("%s", e.strerror)
                    if not self.reconnect:
                        sys.exit(1)
                    self.logger.info("Trying to reconnect to {} in 10 sec"
                                     .format(self.port))
                    time.sleep(10)
                except OSError as e:
                    self.logger.error("Cannot connect to serial port {}: {}"
                                      .format(self.port, e.strerror))
                    sys.exit(1)
```

Reply:
Mh... when I do that and have no device connected it then tries to reconnect. This is not expected behavior.

Reply:
Fixed.